packages/example/pages/index.tsx

```diff
@@ -159,95 +160,132 @@ const Example = ({ authEnabled }: AppContextProps) => {
+                          {account.displayName}
+                          {account.displayBalance
+                            ? ` (${account.displayBalance})`
+                            : ''}
+                        </button>
+                      </div>
                     );
-                  }
+                  })()}
+                </div>
+              );
+            }}
+          </ConnectButton.Custom>
 
-                  return (
-                    <div style={{ display: 'flex', gap: 12 }}>
-                      <button
-                        onClick={openChainModal}
-                        style={{ alignItems: 'center', display: 'flex' }}
-                        type="button"
-                      >
-                        {chain.hasIcon && (
-                          <div
-                            style={{
-                              background: chain.iconBackground,
-                              borderRadius: 999,
-                              height: 12,
-                              marginRight: 4,
-                              overflow: 'hidden',
-                              width: 12,
-                            }}
-                          >
-                            {chain.iconUrl && (
-                              <img
-                                alt={chain.name ?? 'Chain icon'}
-                                src={chain.iconUrl}
-                                style={{ height: 12, width: 12 }}
-                              />
-                            )}
-                          </div>
-                        )}
-                        {chain.name ?? chain.id}
-                      </button>
+          <RainbowButton.Custom>
+            {({ ready, connect }) => {
+              return (
+                <button
+                  type="button"
+                  disabled={!ready}
+                  onClick={connect}
+                  style={{ marginLeft: '18px' }}
+                >
+                  Connect Rainbow
+                </button>
+              );
+            }}
+          </RainbowButton.Custom>
+        </div>
+      </div>
 
-                      <button onClick={openAccountModal} type="button">
-                        {account.displayName}
-                        {account.displayBalance
-                          ? ` (${account.displayBalance})`
-                          : ''}
-                      </button>
-                    </div>
-                  );
-                })()}
-              </div>
-            );
+      <div>
+        <h3 style={{ fontFamily: 'sans-serif' }}>Rainbow button</h3>
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            gap: 10,
           }}
-        </ConnectButton.Custom>
+        >
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'column',
+            }}
+          >
+            <RainbowButton />
```

Comment:
We should add `<WalletButton wallet='metaMask'/>` here to test the stateful behavior of two buttons alongside each other (which some devs might do for their wallet list)

Reply:
Perfect feedback ⚡️. Looking forward for this feature where you can pass any provider ✨